SYCL/ESIMD/double_grf.cpp

```diff
@@ -0,0 +1,193 @@
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// This test verifies effect of
+//   set_kernel_properties(kernel_properties::use_double_grf);
+// API call in device code:
+// - ESIMD/SYCL splitting happens as usual
+// - ESIMD module is further split into callgraphs for entry points requesting
+//   "double GRF" and callgraphs for entry points which are not
+// - ESIMD device binary images requesting "double GRF" must be compiled with
+//   -doubleGRF option
+
+// REQUIRES: gpu-intel-pvc
+// UNSUPPORTED: cuda || hip
+// TODO/FIXME: esimd_emulator does not support online compilation that
+//             invokes 'piProgramBuild'/'piKernelCreate'
+// UNSUPPORTED: esimd_emulator
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: env SYCL_PI_TRACE=-1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 %GPU_CHECK_PLACEHOLDER --check-prefixes=CHECK,CHECK-NO-VAR
+// RUN: env SYCL_PROGRAM_COMPILE_OPTIONS="-g" SYCL_PI_TRACE=-1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 %GPU_CHECK_PLACEHOLDER --check-prefixes=CHECK,CHECK-WITH-VAR
+
+#include "esimd_test_utils.hpp"
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <sycl/ext/intel/esimd.hpp>
+
+using namespace cl::sycl;
+using namespace sycl::ext::intel::esimd;
+using namespace sycl::ext::intel::experimental::esimd;
+
+bool checkResult(const std::vector<float> &A, int Inc) {
+  int err_cnt = 0;
+  unsigned Size = A.size();
+
+  for (unsigned i = 0; i < Size; ++i) {
+    if (A[i] != i + Inc)
+      if (++err_cnt < 10)
+        std::cerr << "failed at A[" << i << "]: " << A[i] << " != " << i + Inc
+                  << "\n";
+  }
+
+  if (err_cnt > 0) {
+    std::cout << "  pass rate: "
+              << ((float)(Size - err_cnt) / (float)Size) * 100.0f << "% ("
+              << (Size - err_cnt) << "/" << Size << ")\n";
+    return false;
+  }
+  return true;
+}
+
+// Make the double GRF request from non-inlineable function - compiler should
+// mark the caller kernel as "double GRF" anyway.
+__attribute__((noinline)) void double_grf_marker() {
+  set_kernel_properties(kernel_properties::use_double_grf);
+}
+
+int main(void) {
+  constexpr unsigned Size = 32;
+  constexpr unsigned VL = 16;
+
+  std::vector<float> A(Size);
+
+  for (unsigned i = 0; i < Size; ++i) {
+    A[i] = i;
+  }
+
+  try {
+    buffer<float, 1> bufa(A.data(), range<1>(Size));
+    queue q(gpu_selector{}, esimd_test::createExceptionHandler());
+
+    auto dev = q.get_device();
+    std::cout << "Running on " << dev.get_info<info::device::name>() << "\n";
+
+    auto e = q.submit([&](handler &cgh) {
+      auto PA = bufa.get_access<access::mode::read_write>(cgh);
+      cgh.parallel_for<class SyclKernel>(Size,
+                                         [=](id<1> i) { PA[i] = PA[i] + 1; });
+    });
+    e.wait();
+  } catch (cl::sycl::exception const &e) {
```

Comment:
minor/non-blocking comment:
```suggestion
  } catch (sycl::exception const &e) {
```